### PR TITLE
Support for vault KV v2

### PR DESCRIPTION
**Pull request by Hi-Fi**
Support to generate JWT token from private key

Cryptography and pyjwt to dependencies

### CODE REVIEW

**Comment by Hi-Fi** [comment]
Misses monorepo support, which is reason for being draft

File: Dockerfile [hunk 1]
```diff
@@ -108,11 +108,13 @@ ENV RUSTUP_HOME=/usr/local/rustup \
 RUN apk add --no-cache --virtual .build-deps \
         build-base \
         python3-dev \
+        gcc \
```
Comment (frwickst) [inline Dockerfile]: No need for these two, since `build-base` is already a meta package that includes them both .

File: Dockerfile [hunk 2]
```diff
@@ -108,11 +108,13 @@ ENV RUSTUP_HOME=/usr/local/rustup \
...
+        musl-dev \
     && apk add --no-cache --virtual .fetch-deps \
         curl \
     && apk add --no-cache \
         python3 \
-        bash \
+        bash \
```
Comment (frwickst) [inline Dockerfile]: Excessive whitespace please remove

**Review by frwickst — CHANGES_REQUESTED** [review]
A couple of things that I noticed.

Please remember to run the suite of styling tests as well, the easiest way to do this is to just use `pre-commit` and have it do all the checks for you for each commit. You can install following these instructions https://pre-commit.com/ (basically install the software, then run `pre-commit install`)

File: kolga/libs/vault.py [hunk 3]
```diff
@@ -34,9 +37,20 @@ def __init__(
                 message="VAULT_ADDR not defined. Skipping Vault usage.",
             )
 
-    def login(self, ci_jwt: str = settings.VAULT_JWT) -> None:
+    def login(self, ci_jwt: str = settings.VAULT_JWT, ci_jwt_private_key: str = settings.VAULT_JWT_PRIVATE_KEY) -> None:
         if self.initialized:
             try:
+                if ci_jwt_private_key:
+                    ci_jwt = encode(
```
Comment (frwickst) [inline kolga/libs/vault.py:48]: MyPy will not like this since `encode()` produces `bytes` and `ci_jwt` is set to `str`.

Comment (Hi-Fi) [reply kolga/libs/vault.py:48]: It doesn't seem to do so:
```
type(jwt_token)
<class 'str'>
```

https://github.com/jpadilla/pyjwt/issues/391
https://github.com/jpadilla/pyjwt/commit/e3a1ba52b5926bd419a5bab73c64d7bce1089958
https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py#L37-L44

File: Dockerfile (hunk 2 above)
Comment (Hi-Fi) [reply Dockerfile]: Ah, apparently that removal was missing from `.editorconfig`

File: Dockerfile (hunk 1 above)
Comment (Hi-Fi) [reply Dockerfile]: Removed. Made some tests with base image and just copied those directly from those tries and/or cryptography installation guide.

**Comment by Hi-Fi** [comment]
Monorepos should be handled in pipeline by defining project name correct way (and by creating correctly named secrets to Vault).

**Review by joonvena — APPROVED** [review]
LGTM

**Pull request merged**